modal/client.py

```diff
@@ -161,7 +162,7 @@ async def _close(self, prep_for_restore: bool = False):
 
     async def _init(self):
         """Connect to server and retrieve version information; raise appropriate error for various failures."""
-        logger.debug("Client: Starting")
+        logger.debug(f"Client ({id(self)}): Starting")
```

Comment:
Having the object ID logged helps track the usage of fresh vs. stale clients when debugging snapshot issues. 

Instead of having to catch stale `client` objects and refresh them it'd be better if the client object itself could catch that it was stale and refresh itself. If this could work then we could remove all current (and future) `if self.client._snapshotted` type checks.

Reply:
Yeah I agree, we should make the client itself detect this, which should be doable through the `_call_unary` and `_call_stream` methods on it which all RPC methods should be going via now